Sources/PackageDescription4/SupportedPlatforms.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2018 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+public struct SupportedPlatform: Encodable {
+
+    let platform: String
+    let version: VersionedValue<String>?
+
+    init(platform: String, version: VersionedValue<String>? = nil) {
+        self.platform = platform
+        self.version = version
+    }
+
+    public static func macOS(_ version: SupportedPlatform.MacOSVersion) -> SupportedPlatform {
+        return SupportedPlatform(platform: "macosx", version: version.version)
+    }
+
+    public static func iOS(_ version: SupportedPlatform.MacOSVersion) -> SupportedPlatform {
+        return SupportedPlatform(platform: "iphoneos", version: version.version)
+    }
+
+    public static func tvOS(_ version: SupportedPlatform.MacOSVersion) -> SupportedPlatform {
+        return SupportedPlatform(platform: "appletvos", version: version.version)
+    }
+
+    public static func watchOS(_ version: SupportedPlatform.MacOSVersion) -> SupportedPlatform {
+        return SupportedPlatform(platform: "watchos", version: version.version)
+    }
+
+    public static func linux() -> SupportedPlatform {
+        return SupportedPlatform(platform: "linux")
+    }
+
+    public static var all: SupportedPlatform {
+        return SupportedPlatform(platform: "<all>")
+    }
+}
+
+extension SupportedPlatform {
+    /// The macOS version.
+    public struct MacOSVersion: Encodable {
+
+        /// The underlying version representation.
+        let version: VersionedValue<String>
+
+        private init(_ version: String, supportedVersions: [ManifestVersion]) {
+            let api = "v" + version.split(separator: ".").joined(separator: "_")
+            self.version = VersionedValue(version, api: api, versions: supportedVersions)
+        }
+
+        public static let v10_10: MacOSVersion = .init("10.10", supportedVersions: [.v5])
```

Comment:
We should have values for every released SDK, not just each released "major" version. Those are:

- 10.10
- 10.10.2
- 10.10.3 
- 10.11
- 10.11.2 
- 10.11.3
- 10.11.4 
- 10.12 
- 10.12.1 
- 10.12.2 
- 10.12.4 
- 10.13
- 10.13.1 
- 10.13.2 
- 10.13.4 
- 10.14 
- 10.14.1

Reply:
We're planning to do only major versions (see proposal)

Reply:
Nothing in the proposal seems to explicitly indicate this, the closest thing being:

> This could be because the new version was recently released **or isn't appropriate** to be included in the APIs (for e.g. **dot versions**).

...but doesn't explain why it makes the assertion that dot versions are for some reason inappropriate to be included in the APIs.

Reply:
I guess a good option would be having parity with SDK's `DEPLOYMENT_TARGET_SUGGESTED_VALUES` values.